### ISSUE TITLE
Subdocuments get overwritten

### ISSUE THREAD

**Issue by relief-melone** [Automattic/mongoose#8044]
First of all I am not entirely sure if this is a bug or expected behaviour. Because of this I also have a Question open at stackoverflow
https://stackoverflow.com/questions/57324321/mongoose-overwriting-data-in-mongodb-with-default-values-in-subdocuments/57324552#57324552

## Bug

**Current Behaviour**
When updating documents with objects missing certain keys findOneAndUpdate results in only updating the fields supplied. However if you update with a missing subdocument the value of this subdocument gets set to its default value

**Expected Behaviour**
If no Subdocument is supplied it should be left as is on the database

**Steps to reproduce**

1. Create a Schema with a subdocument
```typescript
const someSchema:Schema = new mongoose.Schema({
  Title: String,
  Subdocuments: [{
    SomeValue: String
    Position: {
     X: {type: Number, default: 0},
     Y: {type: Number, default: 0},
     Z: {type: Number, default: 0}
    }
  }]  
});
```
2. Create an entry in the database with values for that subdocument
3. Update the document using findOneAndUpdate (though I guess this will apply to other update methods as well) and leave out a subdocument (in this case Position)

The result will be an updated Document with the Position of that document being updated to X:0 Y:0 and Z:0. Heres the code for the update method

```typescript
const updateById = async (Id: string, NewDoc: DocClass) => {
    let doc: DocClass | null = await DocumentModel.findOneAndUpdate(
        { _id: Id },
        { $set: NewDoc },
        { new: true, runValidators: true });
    if (!doc) {
        throw createError.documentNotFound(
          { msg: `The Document you tried to update (Id: ${Id}) does not exist` }
        );
    }
    return doc;
}
```

**Versions**
MongoDB: 4.2
Mongoose: 5.6.4
Node.js: 10.15.3

**Comment by relief-melone** [comment]
What I guess by now is that actually by sending the subdocuments with the id mongoose is not acutally updating the subdocuments but initalizing new ones but setting the id to the old value. which is fine for the get requests from the frontend but ends up running through the create logic and thus setting all the default values for values that are unset

**Comment by vkarpov15** [comment]
What does `NewDoc` look like in your code? Maybe try updating `Subdocuments.0.SomeString` if you want to update a property in place without overwriting the whole subdocument?

**Comment by relief-melone** [comment]
I answered my Question on stackoverflow. Basically Mongoose is doing the correct thing. As I send an Array of Subdocuments with my Document, mongoose will interpret all of those as new instances of a new subdocument. And this is correct as for an update I would have to supply a query. Now I could update each of those subdocuments individually but that would also mean that i have to make a seperate method to sort out which subdocuments in the array have been added, removed, moved etc. 

More detail on StackOverflow. In short: This one can be closed.

However it would be a nice feature if mongoose could somehow handle such arrays and decide which subdocuments are already there and just need to be updated, which ones need to be removed and which ones are added
